src/ChannelServer/World/SkillHandler.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Melia.Channel.Network;
+using Melia.Shared.Data.Database;
+using Melia.Shared.World;
+
+namespace Melia.Channel.World
+{
+	interface ITargetedSkillHandler
+	{
+		void Handle(Skill skill, Character caster, IEntity target);
+	}
+
+	interface IGroundSkillHandler
+	{
+		void Handle(Skill skill, Character caster, Position position);
+	}
+
+	interface ITargetGroundSkillHandler
+	{
+		void Handle(Skill skill, Character caster, Position position, List<IEntity> targets);
+	}
+
+	class TargetedSkillHandler : ITargetedSkillHandler
+	{
+		public void Handle(Skill skill, Character caster, IEntity target)
+		{
+			if (skill.Data.OverHeat != 0)
+				skill.OverHeatCount++;
+			Send.ZC_OVERHEAT_CHANGED(caster, skill);
+			if (skill.Data.OverHeat == 0 || skill.Data.OverHeat == skill.OverHeatCount)
+			{
+				Send.ZC_COOLDOWN_CHANGED(caster, skill);
+				skill.OverHeatCount = 0;
+			}
+
+			var damage = caster.GetRandomPAtk() + 100;
```

Comment:
Is the attack damage still a problem?^^ Maybe we should take a look at the calculations and see what's wrong there. `MinPAtk` and `MaxPAtk` might be outdated or buggy.